Provide fixes for Timer and LowPowerTimer tests.

mprse opened the pull request:
## Description

- LowPoterTimer test gives sometimes failed result while testing measured time accumulation. The check which verifies if total number of elapsed milliseconds is valid fails. Test assumes that delta value equal to 1 ms is sufficient for all test cases, which is not true since in case where time measurement is performed few times in sequence the measurement error also accumulates and 1 ms might be not enough. To solve this problem delta value for milliseconds tests must be updated.
- Since Timer test location is invalid it has been moved to more suitable location: TESTS/mbed_drivers/. Additionally fixed invalid comments and used more suitable ASERT macros.
- Provide fix for Issue https://github.com/ARMmbed/mbed-os/issues/5468. Increased DELTA value.

## Status
READY


## Migrations
NO

0xc0170 commented:
From the commit msg:

```
Additional changes:
- provide fix also for Timer test.
- move Timer test to TESTS/mbed_drivers/ directory which is more suitable.
- fix minor issues and comments.

The check if number of elapsed mili seconds simetimes fails. Test assumes that
```

Seems like msg was truncated, is that correct? Why are you squashing these 3 changes? I do not see which one is bugfix and which one are fix comments / minor issues (what are minor issues) ? It might make sense to split this into at least 2 commits.

mprse commented:
> Seems like msg was truncated, is that correct? Why are you squashing these 3 changes? I do not see which one is bugfix and which one are fix comments / minor issues (what are minor issues) ? It might make sense to split this into at least 2 commits.

Fixed broken commit message - sorry for that. 
Added more detailed information about the changes.
This PR provides fix, but additionally I have also changed few comments and used more suitable ASERT macros. This are these "minor issues" which were catch during the update. Do I need a special commit for this?

mprse commented:
Spitted into 2 commits as suggested and added additional commit with fix for issue https://github.com/ARMmbed/mbed-os/issues/5468.

0xc0170 commented:
/morph build

0xc0170 commented:
/morph test

0xc0170 commented:
/morph test

0xc0170 commented:
@mprse Test passed, this should be good to go!
